notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -0,0 +1,1127 @@
+                    params_dict[var+'_'+alt_name_dic[alt]]['parameters'] = list(BestDict.items())[0][1]
+
+    # Trip Specific Variable (maybe combine with individual specific variables)
+    for var in trip_spec:
+        trip_var = pd.Series([(data_long.loc[data_long[obs_id_col] == x][var].unique()[0]) for x in data_long[obs_id_col].unique()])
+        if var_types[var] == 'categorical':
+            if len(trip_var.unique()) == 1:
+                params_dict[var]['distribution'] = 'constant'
+                params_dict[var]['parameters'] = trip_var.unique()
+            else:
+                params_dict[var]['distribution'] = 'categorical'
+                np_array_range = np.arange(trip_var.max()+1)
+                array_bincount = np.bincount(trip_var)
+                probs = array_bincount / len(trip_var)
+                params_dict[var]['parameters'] = [np_array_range,
+                                                  probs]
+        else:
+            if len(trip_var.unique()) == 1:
+                params_dict[var]['distribution'] = 'constant'
+                params_dict[var]['parameters'] = trip_var.unique()
+            else:
+                fitter_object = Fitter(data=trip_var,
+                                       distributions=cont_dists,
+                                       timeout=30)
+                fitter_object.fit()
+                BestDict = fitter_object.get_best()
+                params_dict[var]['distribution'] = list(BestDict.items())[0][0]
+                params_dict[var]['parameters'] = list(BestDict.items())[0][1]
+
+    return params_dict
+
+
+def SimDf(params_dict, size=1000):
+    """Funtion to simulate data of size N based on specified
+    distribution/parameters found by the fitter package.
+    Inputs:
+    -------
+    data: dataframe from which columns are to be taken
+    dist_params: the distribution parameters from find_dist_df
+    Outputs:
+    -------
+    DataFrame object with simulated data based on specified distributions
+    """
+    Sim_Df = pd.DataFrame(columns=list(params_dict.keys()))
+    Sim_Df = Sim_Df.fillna(0)
+    for column in list(params_dict.keys()):
+        if params_dict[column]['distribution'] == 'categorical':
+            data_sim = np.random.choice(a=params_dict[column]['parameters'][0],
+                                        p=params_dict[column]['parameters'][1],
+                                        size=size)
+            Sim_Df[column] = data_sim
+        elif params_dict[column]['distribution'] == 'constant':
+            data_sim = params_dict[column]['parameters'][0]
+            Sim_Df[column] = data_sim
+        else:
+            dist = getattr(scipy.stats, params_dict[column]['distribution'])
+            data_sim = dist.rvs(*params_dict[column]['parameters'], size=size)
+            Sim_Df[column] = data_sim
+    return Sim_Df
+
+
+def SimulateAvailability(data_long, sim_data, obs_id_col, alt_name_dict):
```

Comment:
What does this function do? Can you put a function description in the function's docstring?

Reply:
And fear not, it's. clear what your function does from looking at the code =).

Reply:
@timothyb0912 check out [71e7e92](https://github.com/hassanobeid1994/tr_b_causal_2020/pull/44/commits/71e7e9299542cdfee41485f9f56da73fa60d9852)